crates/bevy_app/src/app.rs

```diff
@@ -203,85 +201,89 @@ impl App {
     /// App::new()
     ///     .add_system(my_system);
     /// ```
-    pub fn add_system<Params>(&mut self, system: impl IntoSystemDescriptor<Params>) -> &mut Self {
-        self.add_system_to_stage(CoreStage::Update, system)
-    }
-
-    pub fn add_system_set(&mut self, system_set: SystemSet) -> &mut Self {
-        self.add_system_set_to_stage(CoreStage::Update, system_set)
-    }
-
-    pub fn add_system_to_stage<Params>(
-        &mut self,
-        stage_label: impl StageLabel,
-        system: impl IntoSystemDescriptor<Params>,
-    ) -> &mut Self {
-        self.schedule.add_system_to_stage(stage_label, system);
-        self
-    }
-
-    pub fn add_system_set_to_stage(
-        &mut self,
-        stage_label: impl StageLabel,
-        system_set: SystemSet,
-    ) -> &mut Self {
-        self.schedule
-            .add_system_set_to_stage(stage_label, system_set);
+    pub fn add_system<Params>(&mut self, system: impl IntoSystem<(), (), Params>) -> &mut Self {
+        let system = system.system();
+        if system.config().startup {
+            let system = {
```

Comment:
I think this is required because we need to access config and unless we call the .system() we can't actually access it.

Reply:
I just mean the block for 
```rust
let system = {
	if system.config().stage.is_some() {
    	system
  	} else {
      	system.stage(StartupStage::Startup)
	}
}
//vs
let system = if system.config().stage.is_some() {
	system
} else {
	system.stage(StartupStage::Startup)
};
```
I tried it locally and it seems to work fine

Reply:
@thechubbypanda I'm pinging you because I don't think you'll see this otherwise since it's resolved. 

Also, this isn't a required change or anything I was just curious if there was a reason for this or if it's just a style preference.